core/2d/FastTMXLayer.cpp

```diff
@@ -908,11 +908,14 @@ void FastTMXLayer::setTileGID(int gid, const Vec2& tileCoordinate, TMXTileFlags
 
 void FastTMXLayer::setupTileSprite(Sprite* sprite, const Vec2& pos, uint32_t gid)
 {
-    sprite->setPosition(getPositionAt(pos));
     sprite->setPositionZ((float)getVertexZForPos(pos));
-    sprite->setAnchorPoint(Vec2::ZERO);
     sprite->setOpacity(this->getOpacity());
 
+    // fix issue #1283 too;  put the anchor in the middle for ease of rotation. 
+    sprite->setAnchorPoint(Vec2(0.5f, 0.5f));
+    sprite->setPosition(getPositionAt(pos).x + std::roundf(sprite->getContentSize().height / 2),
```

Comment:
I would suggest to store `sprite->getContentSize()` in a local variable rather than doing two calls.

Same for `getPositionAt(pos)`, there is no need for two calls here :)